src/commands/staff/addassignment.ts

```diff
@@ -1,24 +1,42 @@
-import { EmbedField, Message, MessageEmbed } from 'discord.js';
+import { ApplicationCommandOptionData, ApplicationCommandPermissionData, CommandInteraction, EmbedField, Message, MessageEmbed } from 'discord.js';
 import { Course } from '@lib/types/Course';
-import { staffPerms } from '@lib/permissions';
+import { ADMIN_PERMS, staffPerms, STAFF_PERMS } from '@lib/permissions';
 import { DB } from '@root/config';
 import { Command } from '@lib/types/Command';
 
 export default class extends Command {
 
 	// Never assume staff are not dumb (the reason this is so long)
 
+	tempPermissions: ApplicationCommandPermissionData[] = [STAFF_PERMS, ADMIN_PERMS];
+
 	description = 'Adds an assignment to a given course ID\'s assignment list';
 	usage = '<course ID>|<assignmentID(s)>';
 	runInDM = false;
-	aliases = ['adda'];
+
+	options: ApplicationCommandOptionData[] =[
+		{
+			name: 'course',
+			description: 'The course ID to add an assignment to',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'newassignments',
+			description: 'A comma separated list of new assignments',
```

Comment:
Would suggest using | instead of commas to separate items in case assignment names use commas themselves. (It happens - assumue profs are like that :P)